src/mixins/filing-template-mixin.ts

```diff
@@ -290,18 +292,27 @@ export default class FilingTemplateMixin extends Mixins(DateMixin, EnumMixin) {
         identifier: this.getEntitySnapshot.businessInfo.identifier,
         legalName: this.getEntitySnapshot.businessInfo.legalName,
         legalType: this.getEntitySnapshot.businessInfo.legalType,
-        nrNumber: this.getEntitySnapshot.businessInfo.nrNumber,
-        associationType: this.getAssociationType
+        nrNumber: this.getEntitySnapshot.businessInfo.nrNumber
       },
       specialResolution: {
-        ...this.getcreateResolution
+        ...this.getCreateResolution
+      },
+      alteration: {
+        business: {
+          identifier: this.getBusinessId,
+          legalType: this.getEntityType
+        },
+        cooperativeAssociationType: this.getAssociationType,
+        rulesFileKey: 'test',
+        rulesFileName: 'testUrl',
+        memorandumFileKey: 'test',
+        memorandumFileName: 'test'
       }
-
     }
 
     // Apply NR / business name / business type change to filing
     if (this.getNameRequestNumber || this.hasBusinessNameChanged || this.hasBusinessTypeChanged) {
-      filing.alteration.nameRequest = this.getNameRequest
+      filing.changeOfName = { nameRequest: this.getNameRequest }
```

Comment:
Another object at the top level of the filing?

Reply:
Payload:

```
{
    "filing": {
        "header": {
            "name": "specialResolution",
            "date": "2022-08-02",
            "certifiedBy": "test test"
        },
        "business": {
            "legalType": "CP",
            "identifier": "CP1002552",
            "legalName": "TRAV COOP 2",
            "foundingDate": "2022-05-09T00:00:00+00:00"
        },
        "specialResolution": {
            "signatory": {
                "givenName": "givenName",
                "familyName": "familyName",
                "additionalName": "additionalName"
            },
            "resolution": "None",
            "signingDate": "2022-08-02",
            "resolutionDate": "2022-08-02",
            "resolutionConfirmed": true
        },
        "alteration": {
            "business": {
                "legalType": "CP"
            },
            "cooperativeAssociationType": "CSC",
            "rulesFileKey": "test",
            "rulesFileName": "testUrl",
            "memorandumFileKey": "test",
            "memorandumFileName": "test"
        },
        "changeOfName": {
            "nameRequest": {
                "legalType": "CP",
                "legalName": "SUPER COOOP COOP",
                "nrNumber": "NR 4192073",
                "expiry": "2022-10-25T06:59:00+00:00",
                "status": "APPROVED",
                "requestType": "CHG",
                "applicant": {
                    "fullName": "BCREGTEST Bena TEST",
                    "fullAddress": "3545 Winter Dr, Ottawa, ON, A5Z 2B8, CA",
                    "phoneNumber": "2506463431",
                    "emailAddress": "hallo@gmail.com"
                }
            }
        }
    }
}
```

Reply:
Yes, exactly -- why are there 3 sub-objects (not including header and business)? I think all other filings have a single sub-object (named according to `header.name`).

Reply:
I mean, this could be "3 filings in one", but I haven't seen this before.

Reply:
yes, this is 3 filings in one.  it won't always have 3 filings but it's possible depending on what was changed.

![image](https://user-images.githubusercontent.com/6685223/187485148-9e0bf9c2-d422-486e-9074-04daff3d9766.png)

Reply:
I think the SR filing will always exist along with at least one of CoN or alteration.  Unless it's a change that doesn't require a filing.

Reply:
I see! I definitely tagged the right dev -- thanks Argus!

I see the filing code in this PR always adds the alteration object to the filing. Should it be conditional (like the changeOfName object)?

Reply:
yes, alteration should be conditional.

Reply:
if alteration is conditional.. that means the rules and memorandum are optional

Reply:
Is changing rules and memorandum optional for the user?

Reply:
not 100% sure, I think so

We have two more tickets to cover those both.. I think Megan is working on it
We also have another ticket for filing + pay

Reply:
Now that I understand that this is a "multi filing", I'm OK with the multiple filing objects.

I leave it to you to make objects conditional as needed.